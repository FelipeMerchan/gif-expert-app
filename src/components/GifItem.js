import React from 'react';
import PropTypes from 'prop-types';

export const GifItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__fadeIn">
      <img
        className="card__img"
        src={url}
        alt={title}
        title={title}
      />
      <p className="card__title">{title}</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}