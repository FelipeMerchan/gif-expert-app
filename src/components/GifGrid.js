import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';

import { GifItem } from './GifItem';

export const GifGrid = ({ category }) => {

  const { data: images, loading } = useFetchGifs(category);

  return (
    <section aria-label="Gif list">
      <h3 className="animate__animated animate__fadeIn">{category}</h3>

      {loading && <p className="animate__animated animate__flash">Cargando...</p>}

      <div className="card-grid">
        {
          images.map(img => (
            <GifItem
              key={img.id}
              {...img}
            />
          ))
        }
      </div>
    </section>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
