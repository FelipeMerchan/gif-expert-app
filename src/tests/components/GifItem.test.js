import { shallow } from 'enzyme';
import { GifItem } from '../../components/GifItem';

describe('Pruebas en el <GifItem />', () => {
  const title = 'Un titulo';
  const url = 'https://localhost/algo.jpg';
  const wrapper = shallow(<GifItem title={title} url={url}/>);
  test('Debe mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe tener un parrafo con el titulo', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('Debe tener la imagen igual al url y alt de los props', () => {
    const img = wrapper.find('img');
    /* .props() regresa un objeto con las propiedades del elemento */
    expect(img.props().src).toBe(url);
    expect(img.props().alt).toBe(title);
  });

  test('Debe tener la clase animate__fadeIn', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');
    expect(className.includes('animate__fadeIn')).toBe(true);

    /* Otra forma de hacer el test: */
    /* const className = 'animate__fadeIn';
    expect(div.props().className).toContain(className); */
  })
});