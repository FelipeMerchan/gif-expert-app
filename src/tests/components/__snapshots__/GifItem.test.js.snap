// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifItem /> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    className="card__img"
    src="https://localhost/algo.jpg"
    title="Un titulo"
  />
  <p
    className="card__title"
  >
    Un titulo
  </p>
</div>
`;
